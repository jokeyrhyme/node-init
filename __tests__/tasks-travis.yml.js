/* @flow */
'use strict';

jest.unmock('execa');

const {
  isNeeded,
  lib: { pruneCache, pruneInstall, setOSMatrix, updateNpm },
} = require('../lib/tasks/travis.yml.js');

test('isNeeded()', async () => {
  const result = await isNeeded({ cwd: process.cwd(), scope: '' });
  expect(typeof result).toBe('boolean');
});

test('pruneCache({}, { hasYarn: false })', () => {
  const results = pruneCache({});
  expect(results).toMatchSnapshot();
});

test('pruneInstall({ ... }, { hasYarn: false })', () => {
  const results = pruneInstall(
    {
      install: ['npm install --global yarn'],
    },
    { hasYarn: false },
  );
  expect(results).toMatchSnapshot();
});
test('pruneInstall({ ... }, { hasYarn: true })', () => {
  const results = pruneInstall(
    {
      install: ['npm install --global yarn'],
    },
    { hasYarn: true },
  );
  expect(results).toMatchSnapshot();
});

test('setOSMatrix({ ... }, {})', () => {
  const results = setOSMatrix({ name: 'no-explicit-os' }, {});
  expect(results).toMatchSnapshot();
});
test('setOSMatrix({ ... }, { os: ["freebsd"] })', () => {
  const results = setOSMatrix({ name: 'freebsd' }, { os: ['freebsd'] });
  expect(results).toMatchSnapshot();
});
test('setOSMatrix({ ... }, { os: ["freebsd", "!linux"] })', () => {
  const results = setOSMatrix(
    { name: 'strictly-freebsd' },
    { os: ['freebsd', '!linux'] },
  );
  expect(results).toMatchSnapshot();
});
test('setOSMatrix({ ... }, { os: ["darwin", "win32"] })', () => {
  const results = setOSMatrix(
    { name: 'darwin-win32' },
    { os: ['darwin', 'win32'] },
  );
  expect(results).toMatchSnapshot();
});
test('setOSMatrix({ ... }, { os: ["darwin", "win32", "!linux"] })', () => {
  const results = setOSMatrix(
    { name: 'strictly-darwin-win32' },
    { os: ['darwin', 'win32', '!linux'] },
  );
  expect(results).toMatchSnapshot();
});

test('updateNpm({}, { hasYarn: false })', () => {
  const results = updateNpm({}, { hasYarn: false });
  expect(results).toMatchSnapshot();
});
test('updateNpm({}, { hasYarn: true })', () => {
  const results = updateNpm({}, { hasYarn: true });
  expect(results).toMatchSnapshot();
});
