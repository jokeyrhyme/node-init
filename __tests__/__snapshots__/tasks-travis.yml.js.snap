// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruneCache({}, { hasYarn: false }) 1`] = `Object {}`;

exports[`pruneInstall({ ... }, { hasYarn: false }) 1`] = `Object {}`;

exports[`pruneInstall({ ... }, { hasYarn: true }) 1`] = `
Object {
  "install": Array [
    "npm install --global yarn",
  ],
}
`;

exports[`setOSMatrix({ ... }, { os: ["darwin", "win32", "!linux"] }) 1`] = `
Object {
  "matrix": Object {
    "include": Array [
      Object {
        "os": "osx",
        "osx_image": "xcode10.2",
      },
      Object {
        "os": "windows",
      },
    ],
  },
  "name": "strictly-darwin-win32",
}
`;

exports[`setOSMatrix({ ... }, { os: ["darwin", "win32"] }) 1`] = `
Object {
  "matrix": Object {
    "include": Array [
      Object {
        "os": "osx",
        "osx_image": "xcode10.2",
      },
      Object {
        "os": "windows",
      },
      Object {
        "os": "linux",
      },
    ],
  },
  "name": "darwin-win32",
}
`;

exports[`setOSMatrix({ ... }, { os: ["freebsd", "!linux"] }) 1`] = `
Object {
  "name": "strictly-freebsd",
}
`;

exports[`setOSMatrix({ ... }, { os: ["freebsd"] }) 1`] = `
Object {
  "name": "freebsd",
}
`;

exports[`setOSMatrix({ ... }, {}) 1`] = `
Object {
  "name": "no-explicit-os",
}
`;

exports[`updateNpm({}, { hasYarn: false }) 1`] = `
Object {
  "install": Array [
    "npm install --global npm",
  ],
}
`;

exports[`updateNpm({}, { hasYarn: true }) 1`] = `
Object {
  "install": Array [
    "npm install --global npm",
    "npm install --global yarn",
  ],
}
`;
